server/executor/assertion_runner.go

```diff
@@ -55,13 +58,14 @@ func (e *defaultAssertionRunner) Stop() {
 	}
 }
 
-func (e *defaultAssertionRunner) startWorker(ctx context.Context) {
+func (e *defaultAssertionRunner) startWorker() {
 	for {
 		select {
 		case <-e.exitChannel:
 			fmt.Println("Exiting assertion executor worker")
 			return
 		case assertionRequest := <-e.inputChannel:
+			ctx := e.getCtxFromRequest(assertionRequest)
```

Comment:
nit: I think it would read easier if we have something like `assertionRequest.Context()` and have the func attached to that struct instead.